File: 1337_K_Weakest_Rows_In_A_Matrix.py
class Solution:
    def kWeakestRows(self, mat: List[List[int]], k: int) -> List[int]:
        numberMap = defaultdict(list)
        index = 0 
        for j in mat: 
            numberMap[j.count(1)].append(index)
            index += 1
            
        result = [] 
        for i in sorted(numberMap.keys()):
            while k > 0 and numberMap[i]:
                result.append(numberMap[i].pop(0))
                k -= 1 
        return result 





class Solution:
    def kWeakestRows(self, mat: List[List[int]], k: int) -> List[int]:
        numberMap = defaultdict(list)
        index = 0 
        for j in mat: 
            numberMap[j.count(1)].append(index)
            index += 1 
        result = [] 
        for i in sorted(numberMap.keys()):
            for j in numberMap[i]: 
                result.append(j)
                k -= 1 
                if k == 0:
                    return result 
            if k == 0:
                break
        return result 
        

File: 1337_K_Weakest_Rows_In_A_Matrix.cpp
class Solution {
public:
    vector<int> kWeakestRows(vector<vector<int>>& mat, int k) {
        map<int, vector<int>> x; 
        int index = 0; 
        for(int i = 0; i < mat.size(); i++){
            int z = count(mat[i].begin(), mat[i].end(), 1); 
            x[z].push_back(i);  
        }
        
        // C++ (elements in std::map are ordered by default by the operator >> applied to the key)
        vector<int> result; 
        map<int, vector<int>>::iterator it;
        for(it = x.begin(); it != x.end(); it++){
            while((k > 0) && (!it->second.empty())){
                int l = it->second.front();
                result.push_back(l);
                it->second.erase(it->second.begin()); 
                k--;
            }
        }
        return result; 
    }
};